doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,143 @@
+```
+
+## Deleting file
+
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` CLI flag. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
+Without the flag, if a customer tries to delete, for example test-file.txt, they will get the following error:
+
+```
+$ rm test-file.txt
+rm: cannot remove 'test-file.txt': Operation not permitted
+```
+
+In Mountpoint's logs, following message will be emitted:
+
+```
+WARN unlink{req=8 parent=1 name="test-file.txt"}: 
+mountpoint_s3::fuse: unlink failed: Deletes are disabled. Use '--allow-delete' mount option to enable it.
+```
+
+## Listing file which is shadowed by directory of same name
+
+A file system directory cannot contain both a file and a directory of the same name. If your bucket's directory structure would result in this state, only the directory will be accessible. The object will not be accessible. 
+For more details on how Mountpoint maps S3 object keys to files and directories, see the [semantics documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#mapping-s3-object-keys-to-files-and-directories). 
+For example, our bucket contains the following object keys:
+
+* out
+* out/image.jpg
+
+When listing the content of a directory mounting the bucket in the example above, we will get the following result:
+
+```
+$ ls
+out
+```
+
+Mountpoint logs will show an entry like this:
+
+```
+WARN readdir{req=5 ino=1 fh=2 offset=17}: 
+mountpoint_s3::inode::readdir::ordered:file 'out' (full key "out") is omitted because another directory 'out' exist with the same name.
+```
+
+Although, in S3 express one zone bucket both file and directory will be shown without the above warning being emitted. 
+
+## Renaming a file/directory
+
+Renaming a file or a directory inside the mounted directory is not supported by Mountpoint.
+Attempting to rename will give error like: 
+
+```
+$ mv hello.txt hello_new.txt
+mv: rename hello.txt to hello_new.txt: Function not implemented
+```
+
+Mountpoint logs will show following message for the above error:
+
+```
+rename{req=2 parent=1 name="" newparent=1 newname=""}: mountpoint_s3::fuse: rename failed: operation not supported by Mountpoint
+```
+
+## Accessing Glacier objects
+
+Objects in the Glacier Flexible Retrieval and Glacier Deep Archive storage classes, and the Archive Access and Deep Archive Access tiers of S3 Intelligent-Tiering, are not accessible with Mountpoint. To access these objects with Mountpoint, restore or copy them to another storage class first.
+When listing or trying to access objects in these storage classes, Mountpoint logs will show entries like:
+
+```
+WARN readdirplus{req=10 ino=1 fh=1 offset=0}: 
+mountpoint_s3::inode: objects in the GLACIER and DEEP_ARCHIVE storage classes are only accessible if restored
+```
```

Comment:
Why do we warn on `readdirplus`? Should we fix that?

Reply:
This error is propagated from `is_readable()` [method](https://github.com/awslabs/mountpoint-s3/blob/92aec0a0132aaf8d20093943cb40269bb57ffb2f/mountpoint-s3/src/inode.rs#L1488-L1503) implemented on InodeStat.

Reply:
Ah I see... I'd probably just include one of these two logs since they are very similar.